automated/android/noninteractive-tradefed/tradefed.sh

```diff
@@ -97,15 +97,36 @@ fi
 sleep 60
 
 SERVER="www.google.com"
-info_msg "device-${ANDROID_SERIAL}: About to check with ping ${SERVER}..."
+DNS_TEST_SERVER="8.8.8.8"
+
+info_msg "device-${ANDROID_SERIAL}: Checking network connectivity by pinging ${SERVER}..."
+
+# Try pinging www.google.com first
 if adb shell 'ping -c 10 '"${SERVER}"'; echo exitcode: $?' | grep -q "exitcode: 0"; then
     report_pass "network-available"
 else
+    info_msg "Ping to ${SERVER} failed, testing DNS by pinging ${DNS_TEST_SERVER}..."
     report_fail "network-available"
-    # print more debug information
-    adb shell ip address || true
-    adb shell ping -c 10 8.8.8.8 || true
-    # to be caught by the yaml file
+
+    # If ping to $DNS_TEST_SERVER succeeds, it's likely a DNS issue
+    if adb shell 'ping -c 10 '"${DNS_TEST_SERVER}"'; echo exitcode: $?' | grep -q "exitcode: 0"; then
+        report_pass "network-ping-dns-server"
+
+        # DNS-specific debug information
+        info_msg "DNS resolution issue suspected; gathering DNS configuration information..."
```

Comment:
It is not clear what this message means here. If ping passes, why do we need to print such message ? and Should we print DNS details in case of failure case on DUT as well as on the host ?

Reply:
the ping at L112 is to ping 8.8.8.8, if it passed while the previous ping domain name(www.google.com) failed, then it should indicate some dns setting issue there. The host(and should be worker) dns setting might be different from the DUT side settings, and the host side should work as far as the tests run to here, but saying that, it should be harmless or helpful for debugging on such failed cases. If you have any suggestions on what host side commands to be run, please let me know.

Reply:
nslookup http://google.com/ might work on the host side.

Reply:
I'm fine leaving this for a followup PR.